src/formatconvert/main.cpp

```diff
@@ -0,0 +1,113 @@
+#include <QDebug>
+#include <QGuiApplication>
+#include <QCommandLineParser>
+
+#include "mapformat.h"
+#include "tilesetformat.h"
+#include "mapreader.h"
+#include "mapwriter.h"
+
+using namespace Tiled;
+
+template <typename Format>
+Format *findFormat(QString const &fileName) {
+  FormatHelper<Format> formatHelper(FileFormat::ReadWrite, "");
+  foreach (Format *format, formatHelper.formats()) {
+    if (format->supportsFile(fileName))
+      return format;
+  }
+  return nullptr;
+}
+
+Map *readMap(QString const &fileName) {
+  MapFormat *format = findFormat<MapFormat>(fileName);
+  if (format) {
+    qDebug() << "Reading " << format->nameFilter();
+    return format->read(fileName);
+  } else {
+    qDebug() << "Reading TMX format";
+    MapReader reader;
+    return reader.readMap(fileName);
+  }
+}
+
+bool writeMap(Map *map, QString const &fileName) {
+  MapFormat *format = findFormat<MapFormat>(fileName);
+  if (format) {
+    qDebug() << "Writing " << format->nameFilter();
+    if (!format->write(map, fileName)) {
+      qFatal("%s", qPrintable(format->errorString()));
+      return false;
+    }
+  } else {
+    qDebug() << "Writing TMX format";
+    MapWriter writer;
+    if (!writer.writeMap(map, fileName)) {
+      qFatal("%s", qPrintable(writer.errorString()));
+      return false;
+    }
+  }
+  return true;
+}
+
+bool writeTileset(SharedTileset tileset, QString const &fileName) {
+  TilesetFormat *format = findFormat<TilesetFormat>(fileName);
+  if (format) {
+    qDebug() << "Writing " << format->nameFilter();
+    if (!format->write(*tileset, fileName)) {
+      qFatal("%s", qPrintable(format->errorString()));
+      return false;
+    }
+  } else {
+    qDebug() << "Writing TSX format";
+    MapWriter writer;
+    if (!writer.writeTileset(*tileset, fileName)) {
+      qFatal("%s", qPrintable(writer.errorString()));
+      return false;
+    }
+  }
+  return true;
+}
+
+int main(int argc, char *argv[]) {
+  QGuiApplication app(argc, argv);
+  QGuiApplication::setApplicationName("formatconvert");
+  QGuiApplication::setApplicationVersion("0.1");
+
+  QCommandLineParser parser;
```

Comment:
This will raise the Qt dependency from 5.1 to 5.2, but I have no problems with that. Actually Travis CI builds are done with Qt 5.2 so there should be no issues. But please do raise the minimum Qt version to 5.2.0 in `tiled.pro`.